tools/webpack/packages.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * External dependencies
+ */
+const LiveReloadPlugin = require( 'webpack-livereload-plugin' );
+const CopyWebpackPlugin = require( 'copy-webpack-plugin' );
+const postcss = require( 'postcss' );
+
+const { join } = require( 'path' );
+
+/**
+ * WordPress dependencies
+ */
+const LibraryExportDefaultPlugin = require( '@wordpress/library-export-default-webpack-plugin' );
+
+const baseDir = join( __dirname, '../../' );
+
+/**
+ * Given a string, returns a new string with dash separators converedd to
+ * camel-case equivalent. This is not as aggressive as `_.camelCase` in
+ * converting to uppercase, where Lodash will convert letters following
+ * numbers.
+ *
+ * @param {string} string Input dash-delimited string.
+ *
+ * @return {string} Camel-cased string.
+ */
+function camelCaseDash( string ) {
+	return string.replace(
+		/-([a-z])/g,
+		( match, letter ) => letter.toUpperCase()
+	);
+}
+
+module.exports = function( env = { environment: 'production' } ) {
+	const mode = env.environment;
+	const suffix = mode === 'production' ? '.min': '';
+
+	const gutenbergPackages = [
+		'a11y',
+		'api-fetch',
+		'autop',
+		'blob',
+		'blocks',
+//		'block-library', // Not on npm yet.
+		'block-serialization-default-parser',
+//		'block-serialization-spec-parser', // No build-module folder.
+		'components',
+		'compose',
+		'core-data',
+		'data',
+		'date',
+		'deprecated',
+		'dom',
+		'dom-ready',
+//		'edit-post', // Not on npm yet.
+		'editor',
+		'element',
+//		'escape-html', // Not on npm yet.
+		'hooks',
+		'html-entities',
+		'i18n',
+//		'is-shallow-equal', // No build-module folder.
```

Comment:
There is not going to be `build-module`, nor `build` folder for this package.

Reply:
We shouldn't assume packages need build-module or build folder or anything. `require` or `import` (or point to the root folder) and let webpack do its magic.

Reply:
My thinking was that we want WordPress to become a mono-repo with these packages in time so hardcoding `build-module` for now is fine.

Also using `@wordpress/[package]` doesn't work because that is also defined as external and then these build files don't contain anything.

Reply:
I think we can have a mono-repo structure (like Gutenberg does) without hardcoding `build-module`. I do see this as an artificial barrier for package inclusion as all packages are not compiled (thus have a `build-module` folder)